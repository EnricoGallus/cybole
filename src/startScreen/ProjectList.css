.project {
    display: flex;
    align-items: center;
    padding: 1rem;
    width: 100%;
    text-decoration: underline;
    cursor: pointer;
}

.project:hover {
    color: darkslateblue;
}

.project .project-detail {
    flex: 1 1 0;
}

.project .project-action {
    display: flex;
    flex-direction: column;
}