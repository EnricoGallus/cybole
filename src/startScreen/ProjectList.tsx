import {Grid} from "@geist-ui/core";
import {Link} from "react-router-dom";

function ProjectList() {

    return (
        <Grid.Container>
            <Grid xs={24}>
                <p className={"display-1"}>Projects</p>
            </Grid>
            <Grid xs={24}>
                <Link to="/projectNew" className={"btn btn-primary"}>New Project</Link>
            </Grid>
        </Grid.Container>
    )
}

export default ProjectList;