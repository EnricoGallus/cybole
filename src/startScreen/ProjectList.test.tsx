import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import ProjectList from './ProjectList';

describe('ProjectList', () => {
    test('renders component', () => {
        const { container } = render(<ProjectList />, {
            wrapper: MemoryRouter,
        });

        expect(container).toMatchSnapshot();
    });
});
