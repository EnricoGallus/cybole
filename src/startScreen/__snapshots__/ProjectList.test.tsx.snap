// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectList renders component 1`] = `
<div>
  <div
    class="jsx-3633143255 item jsx-478091869"
  >
    <div
      class="jsx-3066557810 item xs jsx-3484918754"
    >
      <p
        class="display-1"
      >
        Projects
      </p>
    </div>
    <div
      class="jsx-3066557810 item xs jsx-3484918754"
    >
      <a
        class="btn btn-primary"
        href="/projectNew"
      >
        New Project
      </a>
    </div>
  </div>
</div>
`;
